drivers/sub8_actuator_driver/scripts/actuator_driver.py

```diff
@@ -59,18 +59,19 @@ def __init__(self, port, baud=9600):
         rospy.Service('~actuate', SetValve, self.got_service_request)
         rospy.Service('~actuate_raw', SetValve, self.set_raw_valve)
 
-        rospy.spin()
-        # r = rospy.Rate(1)  # hz
-        # while not rospy.is_shutdown():
-        #     # Heartbeat to make sure the board is still connected.
-        #     r.sleep()
-        #     self.ping()
-        #     if not self.parse_response(None):
-        #         rospy.logwarn("The board appears to be disconnected, trying again in 5 seconds.")
-        #         self.thruster_out_alarm.raise_alarm(
-        #             problem_description='The board appears to be disconnected.'
-        #         )
-        #         rospy.sleep(5)
+        #rospy.spin()
+        # Might comment this whole deal out once it all works for sure.
+        r = rospy.Rate(.2)  # hz
+        while not rospy.is_shutdown():
+            # Heartbeat to make sure the board is still connected.
+            r.sleep()
+            self.ping()
```

Comment:
Matt - it appears to be possible to accidentally send a ping command concurrently with one of the service calls (they are called in seperate threads). I would suggest thread-locking using sub8_ros_tools.threading_helpers

@duggerd noticed this